### ISSUE TITLE
Custom Callback Error message not working in certain scenario

### ISSUE THREAD

**Issue by 4coderz** [rickharrison/validate.js#152]
Hi!
I've been using your validation plugin recently. Its awesome!
But there's a problem which I can't figure out. I'd be glad if you could point out the error (if any in my part).

The callback custom error message is overridden with your error message("An error has occurred with your %s field") when I want to display empty string(setMessage({callback_name}, '')).

Basically the algorithm and error message is supported and displayed correctly in CodeIgnitor form validation rules but not working in your JS.

Thanks in advance! 

-------------------------------------------------CODE---------------------------------------------------------------------
*********************************************************************************************************************
My working code of callback in CodeIgnitor:

public function valid_age($value) {
        $currdate =  new DateTime(date("Y-m-d"));
        $selectdate = '';
        $post = $this->input->post();
        if(!empty($post['usr_dob'])) {
            if($post['usr_dob'] != 'Select DOB') {
                $selectdate = new DateTime($post['usr_dob']);
                $interval = $currdate->diff($selectdate);
                $interval = abs($interval->format('%r%y'));
                if ($value != $interval) {
                    $this->form_validation->set_message('valid_age', 'Your %s does not match with your Date of Birth.');
                    return false;
                }
                else return true;
            }
            else {
                $this->form_validation->set_message('valid_age', ''); //This is the part working correctly!!
                return false;
            }
        }
        else return false;
    }
*********************************************************************************************************************
My similar code with your plugin:

valid.registerCallback('age_check', function(value) {
        var selectDate = document.getElementById("txt_dob").value;
        if(selectDate != null) {
            if(selectDate != 'Select DOB') {
                var currDate = new Date();
                selectDate = new Date(selectDate);
                var reqdAge = DateDiff.inYears(selectDate, currDate);
                if(value != reqdAge) {
                    valid.setMessage('age_check', 'Your %s does not match with your Date of Birth.');
                    return false;
                }
                else return true;
            }
            else {
                valid.setMessage('age_check', '');    //This part not working and replaced with your error!
                return false;
            }
        }
        else {alert("select date is null");
            return false;
        }
    });
*******************************************************************************************************************************
PS: I've initialized and declared everything and every other function and variable... Your plugin is working correctly otherwise, only that error is being replaced!
My current workaround is to display a whitespace instead of empty string in that message(it solves the problem but this is not the hotfix I'm looking for!!! )

**Comment by rickharrison** [comment]
What is your end goal? You want the submission to be prevented but dont want an error displayed?

**Comment by 4coderz** [comment]
Basically, I want to display an empty (null) error message for that part as required by my logic!
As I already told, printing empty string throws a system error of yours which is not displayed by the codeigniter form validation.
My end goal is to model that form validation of ci to JS with your plugin.

Display empty string as a custom error message without the JQ plugin flagging it as an exception and overriding it to your error message of 'An error has occurred in ___ field'. How to prevent displaying of this error message of yours! [My hotfix is to use a space custom error message instead of empty string which is not flagged!]

**Comment by rickharrison** [comment]
Why are you displaying an empty error message? I'm just trying to understand your full use case so I can help diagnose the best way to accomplish it.

**Comment by 4coderz** [comment]
Let me tell you what I did in CI code:
If you look at the corresponding code above that I posted, you will find that my callback_valid_age is checking whether the entered value(passed as a parameter matches the required age after computation). If it is not, print the error else if the entered date is null(basically not entered), display empty error message as there is a separate callback for entering the date which will be handling this error, so no need to print for this field.

Now it works perfectly fine in CI. But for your plugin, that empty string is not printed
